src/internal/middleware.js

```diff
@@ -91,5 +93,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     return task
   }
 
+  sagaMiddleware.setContext = (props) => {
```

Comment:
Just found out this was proposed by myself: https://github.com/redux-saga/redux-saga/issues/713#issuecomment-268750352

Although I think only one of `options.context` or `sagaMiddleware.setContext()` is enough.